### ISSUE TITLE
[OAuth][Keycloak] skip_org_role_sync set to true has no effect

### ISSUE THREAD

**Issue by maxreb** [grafana/grafana#69639]
### 
# What went wrong?

**What happened**:
Everytime I logout and login again the organizations is set to the default organization as a viewer.

**What did you expect to happen**:
As described in the [documentation](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#skip_org_role_sync) I expect that the organizations I, as an Grafana Admin had set, keeps set. Actually I want the organizations to be handled via Keycloak but seems to be [impossible](https://community.grafana.com/t/sso-and-how-to-set-orgid-and-roles/5005)? Or only possible with Grafana Enterprise and Teams?

Edit: When I set the as in the documentation described as deprected `oauth_skip_org_role_update_sync` to true in [auth] then everything works as expected. But will it work for Grafana 10, too?

### 
# How do we reproduce it?

**grafana.ini**:
(Redacted secrets and URIs)
```
[auth]
oauth_auto_login = true
# TODO this works but the user have to click on "Logout" twice. We have to wait for grafana team
# to decide to implement id_token_hint, See https://github.com/grafana/grafana/issues/24643
signout_redirect_url = https://XXX/realms/<REALM>/protocol/openid-connect/logout

[auth.generic_oauth]
enabled = true
client_id = grafana
client_secret = ZZZZ
name = keycloak.auth
scopes = "openid profile email"
auth_url = https://XXX/realms/<REALM>/protocol/openid-connect/auth
token_url = http://keycloak:8080/realms/<REALM>/protocol/openid-connect/token
api_url = http://keycloak:8080/realms/<REALM>/protocol/openid-connect/userinfo
allow_sign_up = true
tls_skip_verify_insecure = true
role_attribute_path = contains(roles[*], 'Admin') && 'Admin' || contains(roles[*], 'Editor') && 'Editor' || 'Viewer'
# TODO we need to find a way to set the organization via the OAuth provider see https://github.com/grafana/grafana/pull/55182
skip_org_role_sync = true
```
**logout**
Then logout and login again.

### 
# What Grafana version are you using?

v9.5.3

### 
### Grafana Platform?

Docker

### Is this a Regression?

None

### Anything else to add?

Using Keycloak v21.0

**Comment by eleijonmarck** [comment]
i will have a look at this and try to reproduce

**Comment by duchymor** [comment]
Hello @eleijonmarck ,
here is another clue as I have a very similar issue (with AzureAD):
when setting it by ENV GF_AUTH_AZUREAD_SKIP_ORG_ROLE_SYNC=true, it is ignored and defaults to false. And I don't even see message "Config overridden from Environment variable" during boot. For now I also stay with the deprecated option.

**Comment by eleijonmarck** [comment]
@duchymor i see. we had an issue with env not overriding this specific setting. 

The fix has been implemented for overriding env variables and is available in 10.0.x.
https://github.com/grafana/grafana/pull/68364

**Comment by AndreasBergmeier6176** [comment]
We have the same effect using azuread. We are using 9.3.11.

Users are set up as Admin/Editor but as soon as they login via AzureAD, their roles are set to "Viewer".

**Comment by eleijonmarck** [comment]
> We have the same effect using azuread. We are using 9.3.11.
> 
> Users are set up as Admin/Editor but as soon as they login via AzureAD, their roles are set to "Viewer".

@AndreasBergmeier6176 and you are setting the `skip_org_role_sync = true`? under the `[auth.azuread]` section.

**Comment by AndreasBergmeier6176** [comment]
> @AndreasBergmeier6176 and you are setting the `skip_org_role_sync = true`? under the `[auth.azuread]` section.

Yes.

**Comment by eleijonmarck** [comment]
@maxreb hey max!

> As described in the [documentation](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#skip_org_role_sync) I expect that the organizations I, as an Grafana Admin had set, keeps set. Actually I want the organizations to be handled via Keycloak but seems to be [impossible](https://community.grafana.com/t/sso-and-how-to-set-orgid-and-roles/5005)? Or only possible with Grafana Enterprise and Teams?

The way to automatically sync the organisation to be handled via keycloak is to use enterprise and team-sync. See our documentation on [team sync for keycloack oauth](https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/keycloak/#teamsync) for info on setting this up.


> Everytime I logout and login again the organizations is set to the default organization as a viewer.

We have been able to replicate the issue and applying a fix in the following PR  - https://github.com/grafana/grafana/pull/72752

**Comment by eleijonmarck** [comment]
@maxreb we have merged the fix and should be available in the 10.1 version and higher.
